From b4148dbeceee4a2fe4e99acdeeb86edc5e4eed01 Mon Sep 17 00:00:00 2001
From: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date: Tue, 29 Aug 2017 16:40:53 -0400
Subject: [PATCH 3/3] Use grub-file to figure out whether multiboot2 should be
 used for Xen.gz

The multiboot2 is much more preferable than multiboot. Especiall
if booting under EFI where multiboot does not have the functionality
to pass ImageHandler.

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
(cherry picked from commit b4d709b6ee789cdaf3fa7a80fd90c721a16f48c2)
---
 util/grub.d/20_linux_xen.in | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index 083bcef5d..0cb0f4e49 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -210,8 +210,13 @@ while [ "x${xen_list}" != "x" ] ; do
 	xen_loader="xen_hypervisor"
 	module_loader="xen_module"
     else
-	xen_loader="multiboot"
-	module_loader="module"
+	if ($grub_file --is-x86-multiboot2 $current_xen); then
+	    xen_loader="multiboot2"
+	    module_loader="module2"
+	else
+	    xen_loader="multiboot"
+	    module_loader="module"
+        fi
     fi
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
-- 
2.24.1

